which-key.el

```diff
@@ -1811,114 +1770,89 @@ ones. PREFIX is for internal use and should not be used."
-                                   (and (re-search-forward "\t+" nil t)
-                                        (match-end 0))))
-                  key binding)
-              (when binding-start
-                (setq key (buffer-substring-no-properties
-                           (point) binding-start))
-                (setq binding (buffer-substring-no-properties
-                               binding-start
-                               (line-end-position)))
-                (save-match-data
-                  (cond
-                   ((member binding ignore-bindings))
-                   ((string-match-p which-key--ignore-keys-regexp key))
-                   ((and prefix
-                         (string-match (format "^%s[ \t]\\([^ \t]+\\)[ \t]+$"
-                                               key-str-qt) key))
-                    (unless (assoc-string (match-string 1 key) bindings)
-                      (push (cons (match-string 1 key)
-                                  (which-key--compute-binding binding))
-                            bindings)))
-                   ((and prefix
-                         (string-match
-                          (format
-                           "^%s[ \t]\\([^ \t]+\\) \\.\\. %s[ \t]\\([^ \t]+\\)[ \t]+$"
-                           key-str-qt key-str-qt) key))
-                    (let ((stripped-key (concat (match-string 1 key)
-                                                " \.\. "
-                                                (match-string 2 key))))
-                      (unless (assoc-string stripped-key bindings)
-                        (push (cons stripped-key
-                                    (which-key--compute-binding binding))
-                              bindings))))
-                   ((string-match
-                     "^\\([^ \t]+\\|[^ \t]+ \\.\\. [^ \t]+\\)[ \t]+$" key)
-                    (unless (assoc-string (match-string 1 key) bindings)
-                      (push (cons (match-string 1 key)
-                                  (which-key--compute-binding binding))
-                            bindings)))))))))
-          (forward-line))
-        (nreverse bindings)))))
+  "Generate a list of current active bindings starting with PREFIX."
+  (let* ((key-desc (concat (key-description prefix) " "))
+         (keys-list (listify-key-sequence prefix))
+         (get-bindings (lambda (map)
+                         (which-key--get-keymap-bindings map nil nil keys-list)))
+         (comparator (lambda (a b)
+                       (string= (car a) (car b))))
+         ;; Get bindings from all active keymaps that match the given prefix.
+         (all-keymaps (mapcar get-bindings (current-active-maps t)))
+         ;; If one of those maps binds <override-state>, then it should shadow others.
+         (binding-is-overriding (lambda (binding)
+                                  (string-suffix-p "<override-state>" (car binding))))
```

Comment:
I'm not sure about the precedence between evil's override state and the overriding keymaps in emacs. I believe the latter has higher preference.

Reply:
Yeah doing a bit more reading, it looks like evil override-state is intended just to override global evil bindings, not any other keymap. So I'm not exactly sure what logic to use here, but probably not what I implemented so far.

Reply:
Great to see this PR. Is this the last remaining issue?